tasks/subscription-manager.yml

```diff
@@ -93,21 +94,14 @@
   when:
     - rhc_state | d("present") in ["present", "reconnect"]
   block:
-    - name: Set the release
+    - name: Set or unset the release
       community.general.rhsm_release:
-        release: "{{ rhc_release }}"
+        release: "{{ rhc_release if rhc_release != __rhc_state_absent
+          else omit }}"
       when:
         - rhc_state | d("present") == "present"
         - __rhc_subman_identity.rc == 0
-        - rhc_release != __rhc_state_absent
-
-    - name: Unset the release
-      community.general.rhsm_release:
-        release: null
-      when:
-        - rhc_state | d("present") == "present"
-        - __rhc_subman_identity.rc == 0
-        - rhc_release == __rhc_state_absent
+        - not rhc_release is none
```

Comment:
in addition to this, most likely you need to check that `rhc_release != omit`; i remember it was needed when the hack for `rhc_baseurl` was needed (see #12); even more so, because
```yaml
  foo: null
...
  rhc_release: "{{ foo | d(omit) }}"
```
will then result in the special `omit-RANDOMSTRING` passed to the role, which means it will be passed to "release:" of `rhsm_release`, and thus unset the release rather than ignoring it - or am i wrong?